src/main/controllers/SubscriptionConfigureListController.ts

```diff
@@ -2,30 +2,45 @@ import { Response } from 'express';
 import { cloneDeep } from 'lodash';
 import { PipRequest } from '../models/request/PipRequest';
 import { SubscriptionService } from '../service/SubscriptionService';
-import { FilterService } from '../service/FilterService';
 
 const subscriptionService = new SubscriptionService();
-const filterService = new FilterService();
 
 export default class SubscriptionConfigureListController {
     public async get(req: PipRequest, res: Response): Promise<void> {
         const listTypes = await subscriptionService.generateListTypesForCourts(
             req.user['userId'],
             req.user['userProvenance'],
-            req.query?.filterValues as string,
-            req.query?.clear as string,
+            null,
+            null,
             req.lng
         );
 
         res.render('subscription-configure-list', {
             ...cloneDeep(req.i18n.getDataByLanguage(req.lng)['subscription-configure-list']),
             listTypes: listTypes['listOptions'],
-            filterOptions: listTypes['filterOptions'],
         });
     }
 
-    public async filterValues(req: PipRequest, res: Response): Promise<void> {
-        const filterValues = filterService.generateFilterKeyValues(req.body);
-        res.redirect(`subscription-configure-list?filterValues=${filterValues}`);
+    public async post(req: PipRequest, res: Response): Promise<void> {
+        const result = await subscriptionService.createListTypeSubscriptionPayload(req.body['list-selections[]']);
+
+        if (Object.values(result).length == 0) {
```

Comment:
We should change the method "subscriptionService.createListTypeSubscriptionPayload()" to return string array and not object so you don't to use Object.values(result) conversion here. After that can just do "if (result.length)"

Reply:
fixed.